pkg/minikube/cluster/cluster.go

```diff
@@ -97,8 +98,10 @@ func StartHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 		return nil, fmt.Errorf("Error configuring authorization on host: %s", err)
 	}
 
-	if err := registration.RegisterHostVM(h, RegistrationParameters); err != nil {
-		return nil, fmt.Errorf("Error registering the VM: %s", err)
+	if !minishiftConfig.InstanceConfig.IsRegister {
```

Comment:
so here we still rely on this global state. We could add the required information to MachineConfig or  wer do the registration separately in a dedicated method.